### DISCUSSION TITLE
✨ UX - Improving skills filter

### DISCUSSION THREAD

**Discussion by BrookJeynes** [SSWConsulting/SSW.People#567] [Ideas]
Hi Everyone,

As per my conversation with Adam, Brady, and Piers, the skills filter on SSW People is intimidating and causes pain when trying to find a skill due to its length.

Currently, the user is able to filter via technologies. This means the list is super long and has a lot of redundant information.

I recorded a video to explain the pain: https://youtu.be/6-7Oy0plfaQ (2 min)

1. Please comment your opinions and ideas below

![image](https://github.com/SSWConsulting/SSW.People/assets/25432120/7cdc584a-0611-47e3-b46c-c85bcd954612)
**Figure: Lots of Azure options**

With the latest change to people, skills are only displayed if they have a relevant marketing page. Read more here https://github.com/SSWConsulting/SSW.People/issues/496.

This means that we can instead display marketing pages as top-level filters which can be expanded into the separate skills for finer control.

<img width="616" alt="image" src="https://github.com/SSWConsulting/SSW.People/assets/25432120/7a7f994e-3a38-4155-a19c-a83fa7fca41c">

**Figure: Mockup of new design**

**Comment by tkapa** [comment]
+1 to this issue.

We could also do a "pill search" idk what it's called. Think LinkedIn skills.
![image](https://github.com/SSWConsulting/SSW.People/assets/17246482/2dafa5de-01be-4b52-b418-bcb7ca2073f8)

**Reply by christoment** [reply]
Was typing this suggestion too but Tylah finished first 😄 

Love the pill search since this will fix my main annoyance in navigating SSW People:
- Scrolling to find my filter item
- Scrolling to untick my 45th selected item down the page
- Having to CTRL+F to find something - might not be intuitive to non-tech people

**Comment by UlyssesMaclaren** [comment]
I like the concept a lot.

some ideas:
1. add a search so you can start typing what you're looking for. aka CTRL F, but an option in the UI to prompt this
2. if your top level is marketing pages, i think that works well, but be aware some techs will need to show up under multiple pages (e.g. .NET would be in web apps and also .net 8 migration)

**Reply by matt-goldman** [reply]
Agree although `ctrl`+`k` is probably a more ubiquitous convention on the web

**Comment by JeanThirion** [comment]
+42 to the issue !
I like @tkapa's suggestion of using pills.

**Comment by bradystroud** [comment]
✨ Love the idea, but there are a few things to figure out 🤔

Can skills have many marketing pages?
If yes, there could be duplicate skills in this list but you probably wont see them.
Do we wanna filter by price?

💡 Since we have made a link between people and skills - we should at a "Meet the experts" button on all consulting pages with a skill with a user.
So if I'm looking at the Azure page I can click the button and it takes me to SSW People with the Azure filter selected.

**Comment by matt-goldman** [comment]
Great idea, agree with the pain. +1 to @UlyssesMaclaren's suggestion and +25 to @tkapa's.

**Comment by MattParkerDev** [comment]
If you are going to do a tree view, it should probably be collapsible, as without it the list is just as long, albeit sorted into related categories. There should also be some indication that list items are selectable

**Comment by SamBWagner** [comment]
Yeah that looks completely bonkers. Definitely agree with making it a tree-view, pill-search or some other form of nested structure. That big flat list is... big.

**Comment by tiagov8** [comment]
A related issue I created a couple of years ago: https://github.com/SSWConsulting/SSW.People/issues/347
